hs_dictionary/views.py

```diff
@@ -34,3 +34,31 @@ def get(self, request, format=None, query=None):
             universities = ['Too many items to list, please continue typing...']
 
         return Response(universities)
+
+
+class ListSubjectAreas(APIView):
+    """
+    View to list all known subject areas in the system
+    """
+
+    @swagger_auto_schema(auto_schema=None)
+    def get(self, request, format=None, query=None):
+        """
+        Return a list of all vocabulary items
+        :return:
+        """
+        terms = request.GET.get('term', '')
+        term_list = terms.split(' ')
+
+        if len(term_list):
+            filtered_sas = SubjectArea.objects.filter(
+                reduce(lambda x, y: x & y, [Q(name__icontains=word) for word in term_list])
+            )
+            subject_areas = [uni.name for uni in filtered_sas]
```

Comment:
Consider values_list with the `flat` parameter.  https://docs.djangoproject.com/en/4.2/ref/models/querysets/#values-list

Reply:
done, thanks for the recommendation!